internal/rbd/rbd_util.go

```diff
@@ -614,6 +620,40 @@ func genVolFromVolumeOptions(ctx context.Context, volOptions, credentials map[st
 	return rbdVol, nil
 }
 
+// validateAndGetImageFeaturesWithMounter validates image features and mounters, and gets their values
+func validateAndGetImageFeaturesWithMounter(volOptions map[string]string) (string, string, error) {
+	mounter, found := volOptions["mounter"]
+	if !found {
+		mounter = rbdDefaultMounter
+	}
+
+	imageFeatures, found := volOptions["imageFeatures"]
+	if !found {
+		return "", mounter, nil
+	}
+
+	arr := strings.Split(imageFeatures, ",")
+	featureSet := sets.NewString(arr...)
+	for _, f := range arr {
+		sf, found := supportedFeatures[f]
+		if !found {
+			return "", "", errors.Errorf("invalid feature %s for volume csi-rbdplugin", f)
```

Comment:
@woohhan just wondering will this cause some backward compatibiliy issues depend on the backend ceph cluster.  That said, suppose a feature called `X` is enabled and in ceph cluster version "Y" and if its working for the customer atm. If this is getting connected to a cluster which is `Y-1` and if `X` is not a supported feature, we will return saying "invalid feature" , Isnt it ?

Reply:
Thanks for the comments! I understood what your saying, but I wonder what scenario does that happen. Are there any situations where a running ceph-csi connects to a lower version Ceph? (for example, Ceph downgrade??)

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
   name: rook-ceph-block
provisioner: rook-ceph.rbd.csi.ceph.com
parameters:
    clusterID: rook-ceph
    // If the version goes down, we must delete this `Y` and re-create the storage class.
    imageFeatures: Y
```

Reply:
@woohhan the point here is, as long as customer is free to connect to any cluster version which is listed in the matrix ( README) at any point in time. So, if SC has a param which is not compatible, I am wondering `returning` error or `just logging` and moving on is the better way. also there is a fact that, may be soon multicluster support..etc may get added .

Reply:
@humblec I didn't think about that. I changed the code with just log when error cases. It will help the user to track the error in case of invalid image features(or mounter)